projects/igniteui-angular/src/lib/combo/combo-dropdown.component.ts

```diff
@@ -205,127 +205,107 @@ export class IgxComboDropDownComponent extends IgxDropDownBase implements OnDest
-                            vContainer.state.startIndex + vContainer.state.chunkSize < vContainer.igxForOf.length) {
-                            vContainer.scrollNext();
-                        }
-                    });
-                    vContainer.scrollNext();
-                    return;
-                }
-            }
-        }
-        if (newIndex === -1) {
-            this.navigateVirtualItem(direction, extraScroll ? 1 : 0);
-        } else if (newIndex === this.lastVisibleIndex && !this.isScrolledToLast) {
-            this.navigateVirtualItem(direction, extraScroll ? 1 : 0);
-        } else if (newIndex === this.lastVisibleIndex && this.isScrolledToLast) {
-            // When initially scrolling to the last item, a pseudo element is present in the children list
-            // We need to check if the element we're on is an actual element or an empty 'igx-combo-item' child
-            if (this.items[newIndex].element && this.items[newIndex].element.nativeElement.clientHeight) {
-                super.navigateItem(newIndex);
-            }
-            return;
+        const allData = vContainer.igxForOf;
+        const extraScroll = this.combo.isAddButtonVisible() ? 1 : 0;
+        const notVirtual = vContainer.dc.instance.notVirtual;
+        if (notVirtual || !direction) { // If list has no scroll OR no direction is passed
+            super.navigateItem(newIndex); // use default scroll
         } else {
-            super.navigateItem(newIndex);
+            if (direction === Navigate.Up) { // Navigate UP
+                this.navigateUp(allData, vContainer, extraScroll, newIndex);
+            } else if (direction === Navigate.Down) { // Navigate DOWN
+                this.navigateDown(allData, vContainer, extraScroll, newIndex);
+            }
         }
     }
 
-    private navigateVirtualItem(direction: Navigate, extraScroll?: number) {
-        const vContainer = this.verticalScrollContainer;
-        // If the data is vitualized, data.length === vContainer.chunkSize, so the below checks are no-longer valid
-        if (vContainer && vContainer.totalItemCount && vContainer.totalItemCount !== 0) {
-            this.navigateRemoteItem(direction);
-            return;
-        }
-        let state = vContainer.state;
-        if (this.isScrolledToLast && direction === Navigate.Down) { // If on the bottom most item, do not subscribe
-            return;
-        }
-        // If on the topmost item, do not subscribe
-        if (this.verticalScrollContainer.getVerticalScroll().scrollTop === 0 && direction === Navigate.Up) {
+    private navigateDown(allData: any[], vContainer: IgxForOfDirective<any>, extraScroll: number, newIndex?: number) {
+        const focusedItem = this.focusedItem;
+        const items = this.items;
+        const children = this.children.toArray();
+        if (focusedItem && (focusedItem.value === allData[allData.length - 1] || focusedItem.value === 'ADD ITEM')) { // If very last item
```

Comment:
Here you can just separate the logic between the last item and add item. First you are checking that it is last or add, then the internal if is checking if it is not add item.

this.combo.isAddButtonVisible() && focusedItem.value === 'ADD ITEM' => return
focusedItem && (focusedItem.value === allData[allData.length - 1] =>this.focusItem(items.length - 1);

Even you can introduce methods similar focusComboSearch for add button, or some of the other code below.